x-pack/plugins/case/server/common/models/commentable_case.ts

```diff
@@ -291,10 +275,33 @@ export class CommentableCase {
       },
     });
 
-    return CollectWithSubCaseResponseRt.encode({
+    const collectionTotalAlerts =
+      countAlertsForID({ comments: collectionComments, id: this.collection.id }) ?? 0;
+
+    const caseResponse = {
       comments: flattenCommentSavedObjects(collectionComments.saved_objects),
-      totalAlerts: countAlertsForID({ comments: collectionComments, id: this.collection.id }),
+      totalAlerts: collectionTotalAlerts,
       ...this.formatCollectionForEncoding(collectionCommentStats.total),
-    });
+    };
+
+    if (this.subCase) {
+      const subCaseComments = await this.service.getAllSubCaseComments({
+        client: this.soClient,
+        id: this.subCase.id,
+      });
+
+      return CaseResponseRt.encode({
+        ...caseResponse,
+        comments: flattenCommentSavedObjects(subCaseComments.saved_objects),
```

Comment:
I'm going to change this. I don't think we want the outer comments to be the subCase comments here. I think this should be `undefined` in the situation where we have the sub case and the parent. The subCase comments should be under `subCases: {comments}`.

Reply:
The comments are needed on the top level from the UI. If you post/patch a comment the case is updated in the UI from the response. If comments are not there the user actions are being shown wrong. Same for the parent.

So if we are on a parent case we return parent's comment. If we are on a subCase, we return subCase's comment.

Reply:
In my opinion we should decide, and improve in another PR, about the response format when we are editing a subCase and when we editing an individual/collection.